lib/livebook/teams/app_deployment.ex

```diff
@@ -0,0 +1,77 @@
+defmodule Livebook.Teams.AppDeployment do
+  use Ecto.Schema
+  alias Livebook.FileSystem
+
+  @file_extension ".zip"
+
+  @type zip_file :: {filename :: String.t(), content :: String.t()}
+  @type zip_files :: list(zip_file())
+
+  @type t :: %__MODULE__{
+          id: String.t() | nil,
+          filename: String.t() | nil,
+          slug: String.t() | nil,
+          sha: String.t() | nil,
+          title: String.t() | nil,
+          deployment_group_id: String.t() | nil,
+          file: binary() | nil,
+          deployed_by: String.t() | nil,
+          deployed_at: NaiveDateTime.t() | nil
+        }
+
+  @primary_key {:id, :string, autogenerate: false}
+  embedded_schema do
+    field :filename, :string
+    field :slug, :string
+    field :sha, :string
+    field :title, :string
+    field :deployment_group_id, :string
+    field :file, :string
+    field :deployed_by, :string
+
+    timestamps(updated_at: nil, inserted_at: :deployed_at)
+  end
+
+  @doc """
+  Creates a new app deployment from session with compressed file.
+  """
+  @spec new(String.t(), String.t(), String.t(), zip_files()) ::
+          {:ok, t()} | {:error, FileSystem.error()}
+  def new(title, slug, deployment_group_id, files) do
+    files =
+      Enum.map(files, fn {filename, content} ->
+        {to_charlist(filename), content}
+      end)
+
+    with {:ok, content} <- zip_files(files),
+         :ok <- validate_size(content) do
```

Comment:
We need to validate the encrypted size, because encryption is going to inflate it a bit, and that's the content Teams API validates on its side.

Reply:
Ur right, makes more sense